src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -632,6 +632,84 @@ def __repr__(self):
             ),
         )
 
+    def __eq__(self, other):
+        """Two grammars are equal if their elements and types are equal.
+
+        NOTE: This could potentially mean that two grammars with
+        the same elements but _different configuration_ will be
```

Comment:
Can you add an example or quick definition of what "configuration" means in this context?

Reply:
Yeah good point - it's basically anything that isn't part of the elements, e.g. `optional`